src/jdk.internal.le/share/classes/jdk/internal/org/jline/terminal/impl/ffm/CLibrary.java

```diff
@@ -0,0 +1,1250 @@
+            cc.put(Attributes.ControlChar.VTIME, (int) c_cc[VTIME]);
+            if (VSTATUS != (-1)) {
+                cc.put(Attributes.ControlChar.VSTATUS, (int) c_cc[VSTATUS]);
+            }
+            // Return
+            return attr;
+        }
+    }
+
+    static MethodHandle ioctl;
+    static MethodHandle isatty;
+    static MethodHandle openpty;
+    static MethodHandle tcsetattr;
+    static MethodHandle tcgetattr;
+    static MethodHandle ttyname_r;
+    static LinkageError openptyError;
+
+    static {
+        // methods
+        Linker linker = Linker.nativeLinker();
+        SymbolLookup lookup = SymbolLookup.loaderLookup().or(linker.defaultLookup());
+        // https://man7.org/linux/man-pages/man2/ioctl.2.html
+        ioctl = linker.downcallHandle(
+                lookup.find("ioctl").get(),
+                FunctionDescriptor.of(
+                        ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.JAVA_LONG, ValueLayout.ADDRESS),
+                Linker.Option.firstVariadicArg(2));
+        // https://www.man7.org/linux/man-pages/man3/isatty.3.html
+        isatty = linker.downcallHandle(
+                lookup.find("isatty").get(), FunctionDescriptor.of(ValueLayout.JAVA_INT, ValueLayout.JAVA_INT));
+        // https://man7.org/linux/man-pages/man3/tcsetattr.3p.html
+        tcsetattr = linker.downcallHandle(
+                lookup.find("tcsetattr").get(),
+                FunctionDescriptor.of(
+                        ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.ADDRESS));
+        // https://man7.org/linux/man-pages/man3/tcgetattr.3p.html
+        tcgetattr = linker.downcallHandle(
+                lookup.find("tcgetattr").get(),
+                FunctionDescriptor.of(ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.ADDRESS));
+        // https://man7.org/linux/man-pages/man3/ttyname.3.html
+        ttyname_r = linker.downcallHandle(
+                lookup.find("ttyname_r").get(),
+                FunctionDescriptor.of(
+                        ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.ADDRESS, ValueLayout.JAVA_LONG));
+        // https://man7.org/linux/man-pages/man3/openpty.3.html
+        LinkageError error = null;
+        Optional<MemorySegment> openPtyAddr = lookup.find("openpty");
+        if (openPtyAddr.isEmpty()) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Unable to find openpty native method in static libraries and unable to load the util library.");
+            List<Throwable> suppressed = new ArrayList<>();
+            try {
+                System.loadLibrary("util");
+                openPtyAddr = lookup.find("openpty");
+            } catch (Throwable t) {
+                suppressed.add(t);
+            }
+            if (openPtyAddr.isEmpty()) {
+                String libUtilPath = System.getProperty("jdk.internal.org.jline.ffm.libutil");
```

Comment:
Wait a minute. If I get this right, then you can pass `-Djdk.internal.org.jline.ffm.libutil=mydodgylib.so` as command line, and have the JDK load any arbitrary library? That sounds super bad from a security perspective. I recommend you remove this.